### PR TITLE
feat(programs): Add User Registry mapping Solana to Light Protocol accounts

### PR DESCRIPTION
**Pull request by vadorovsky**
This registry is the on-chain program where Solana users are supposed to store their Light Protocol public keys. Then this program and its accounts should be used by the SDK for resolution.

### CODE REVIEW

File: light-system-programs/programs/user_registry/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    ) -> Result<()> {
+        let mut user_entry = ctx.accounts.user_entry.load_init()?;
+        user_entry.solana_pubkey = ctx.accounts.signer.key().to_bytes();
+        user_entry.light_pubkey = light_pubkey;
+
+        Ok(())
+    }
+}
+
+#[account(zero_copy)]
```
Comment (ananas-block) [inline light-system-programs/programs/user_registry/src/lib.rs:23]: In this case it's not necessary to use zero_copy but probably doesn't matter.

File: light-system-programs/programs/user_registry/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+pub struct UserEntry {
+    pub solana_pubkey: [u8; 32],
+    pub light_pubkey: [u8; 32],
```
Comment (ananas-block) [inline light-system-programs/programs/user_registry/src/lib.rs:26]: We should either make this 64 bytes or add another 32 bytes field with the encryption pubkey.
You can get the encryption pubkey from the account.
```
let seed32 = bs58.encode(new Uint8Array(32).fill(1));
let k0 = new Account({ poseidon, seed: seed32 });

k0.encryptionKeypair.publicKey
k0.pubkey
// as string
let pubKey = k0.getPublicKey();
```

**Review by ananas-block — CHANGES_REQUESTED** [review]
Looks good just one thing.

File: light-system-programs/programs/user_registry/src/lib.rs (hunk 2 above)
Comment (vadorovsky) [reply light-system-programs/programs/user_registry/src/lib.rs:28]: Done

**Review by ananas-block — APPROVED** [review]

**Pull request merged**